crypto/merkle/hash.go

```diff
@@ -20,7 +21,24 @@ func leafHash(leaf []byte) []byte {
 	return tmhash.Sum(append(leafPrefix, leaf...))
 }
 
+// returns tmhash(0x00 || leaf)
+func leafHashOpt(h hash.Hash, leaf []byte) []byte {
+	h.Reset()
+	h.Write(leafPrefix)
+	h.Write(leaf)
+	return h.Sum(nil)
+}
+
 // returns tmhash(0x01 || left || right)
 func innerHash(left []byte, right []byte) []byte {
 	return tmhash.Sum(append(innerPrefix, append(left, right...)...))
 }
+
+// returns tmhash(0x01 || left || right)
+func innerHashOpt(h hash.Hash, left []byte, right []byte) []byte {
+	h.Reset()
+	h.Write(innerPrefix)
+	h.Write(left)
+	h.Write(right)
+	return h.Sum(nil)
+}
```

Comment:
[optional] can you please include unit test(s) that verifies that the `Opt` functions return a hash identical to the non `Opt` versions. So something like:
- `leafHashOpt()` == `leafHash()`
- `innerHashOpt()` == `innerHash()`

Reply:
Done. No idea why I cannot reply to comments above but:

> [question] What does the suffix Opt mean here?

Named same as in `cometbft` (`opt` for optional 'cause we've `leafHash` or `opt` for optimized).

> // Deprecated: use HashFromByteSlices instead.

Added (+ fixed doc comment).